src/shared/metadata/cgroup_path_resolver_test.cc

```diff
@@ -348,6 +348,46 @@ TEST(LeagcyCGroupPathResolverTest, Cgroup2Format) {
                              ContainerType::kDocker));
 }
 
+// This tests simulates a system where cgroup v1 and v2 are both present. This has been observed
+// in OVH's managed kubernetes service.
+//
+// The testdata/sysfs4-cgroup-v1andv2 directory was created with the following commands:
+// sudo mount -t overlay overlay -o lowerdir=sysfs2/cgroup:sysfs3/cgroup \
+// sysfs4-cgroup-v1andv2/cgroup
+//
+// sysfsv2 and sysfsv3 model cgroupv1 and cgroupv2 respectively.
+TEST(LeagcyCGroupPathResolverTest, Cgroup2With1) {
+  PX_SET_FOR_SCOPE(FLAGS_force_cgroup2_mode, true);
```

Comment:
Why do need this in this case?

Reply:
The reason for this is that the LegacyCGroupPathResolver doesn't operate on a real cgroupv2 filesystem when under test. The newer `CGroupPathResolver` uses `/sys/fs/cgroup` [directly](https://github.com/pixie-io/pixie/blob/c6e18a9817c65d9ad5e350f49612a888266a74c3/src/shared/metadata/cgroup_path_resolver_test.cc#L36) (rather than a `testdata` filesystem mock) meaning this [cgroupv2 base path](https://github.com/pixie-io/pixie/blob/c6e18a9817c65d9ad5e350f49612a888266a74c3/src/shared/metadata/cgroup_path_resolver.cc#L55) can be exercised without the flag. The newer CGroupPathResolver has its template method, which avoids filesystem reads and is why using a `/sys/fs/cgroup` path works.

Reply:
I've removed this test case after more deeply thinking about what test cases we would ideally cover. I've added a TODO explaining those cases in its place.

We have to keep the `FLAGS_test_only_force_cgroup2_mode` flag because that's the only way we can test the LegacyCgroupPathResolver since it actually reads the file filesystem (rather than the template based approach with the CGroupPathResolver).

Reply:
Yeah, a different PR for that refactor is best anyways.